### PR TITLE
store layer in URL and pick locale for initial state

### LINKED ISSUE #118: Store some state in the URL
E.g. for a cycle route it would be nice to select the map layer like it is possible for the old maps with the `layer` parameter:

https://graphhopper.com/maps/?point=51.47212%2C5.970732&point=51.536086%2C5.987549&layer=OpenStreetMap

Then the `key` and `locale` should be picked from the URL but then removed (i.e. not synchronized).

### PR DESCRIPTION
**Pull request by karussell**
This should fix #118

### CODE REVIEW

File: src/NavBar.ts [hunk 1]
```diff
@@ -1,35 +1,40 @@
...
 
-    private onQueryStoreChanged() {
+    private onQueryStateChanged() {
         if (this.isIgnoreQueryStoreUpdates) return
 
         const newHref = QueryUrl.createUrl(
             this.appContext.location.origin + this.appContext.location.pathname,
-            this.queryStore.state
+            this.queryStore.state,
+            this.mapStore.state
```
Comment (karussell) [inline src/NavBar.ts:37]: Not sure where the "root" information of the map layer name should be. I kept it in the MapOptionsStore.

**Comment by Janekdererste** [comment]
This looks good. Although it would result in a conflict with https://github.com/graphhopper/graphhopper-maps/pull/125. I'd rather merge 125 before this one and add the layer information to the new NavBar class

**Comment by karussell** [comment]
Sure :+1:

File: src/NavBar.ts (hunk 1 above)
Comment (Janekdererste) [reply src/NavBar.ts:122]: I think I now know what you mean and I think it belongs into the ```MapOptionsStore```

**Comment by Janekdererste** [comment]
Merging it, since the #125 is merged and I resolved the conflicts

**Pull request merged**